codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/GoWriter.java

```diff
@@ -0,0 +1,798 @@
+     */
+    public GoWriter setDocWrapLength() {
+        this.docWrapLength = DEFAULT_DOC_WRAP_LENGTH;
+        return this;
+    }
+
+    /**
+     * Imports one or more symbols if necessary, using the name of the
+     * symbol and only "USE" references.
+     *
+     * @param container Container of symbols to add.
+     * @return Returns the writer.
+     */
+    public GoWriter addUseImports(SymbolContainer container) {
+        for (Symbol symbol : container.getSymbols()) {
+            addImport(symbol,
+                      CodegenUtils.getSymbolNamespaceAlias(symbol),
+                      SymbolReference.ContextOption.USE);
+        }
+        return this;
+    }
+
+    /**
+     * Imports a symbol reference if necessary, using the alias of the
+     * reference and only associated "USE" references.
+     *
+     * @param symbolReference Symbol reference to import.
+     * @return Returns the writer.
+     */
+    public GoWriter addUseImports(SymbolReference symbolReference) {
+        return addImport(symbolReference.getSymbol(), symbolReference.getAlias(), SymbolReference.ContextOption.USE);
+    }
+
+    /**
+     * Adds and imports the given dependency.
+     *
+     * @param goDependency The GoDependency to import.
+     * @return Returns the writer.
+     */
+    public GoWriter addUseImports(GoDependency goDependency) {
+        dependencies.addAll(goDependency.getDependencies());
+        return addImport(goDependency.getImportPath(), goDependency.getAlias());
+    }
+
+//    /**
+//     * Imports a symbol if necessary using a package alias and list of context options.
+//     *
+//     * @param symbol       Symbol to optionally import.
+//     * @param packageAlias The alias to refer to the symbol's package by.
+//     * @param options      The list of context options (e.g., is it a USE or DECLARE symbol).
+//     * @return Returns the writer.
+//     */
+//    public GoWriter addImport(Symbol symbol, String packageAlias, SymbolReference.ContextOption... options) {
+//        LOGGER.finest(() -> {
+//            StringJoiner stackTrace = new StringJoiner("\n");
+//            for (StackTraceElement element : Thread.currentThread().getStackTrace()) {
+//                stackTrace.add(element.toString());
+//            }
+//            return String.format(
+//                    "Adding Go import %s as `%s` (%s); Stack trace: %s",
+//                    symbol.getNamespace(), packageAlias, Arrays.toString(options), stackTrace);
+//        });
+//
+//        // Always add dependencies.
+//        dependencies.addAll(symbol.getDependencies());
+//
+//        if (isExternalNamespace(symbol.getNamespace())) {
+//            addImport(symbol.getNamespace(), packageAlias);
+//        }
+//
+//        // Just because the direct symbol wasn't imported doesn't mean that the
+//        // symbols it needs to be declared don't need to be imported.
+//        addImportReferences(symbol, options);
+//
+//        return this;
+//    }
```

Comment:
Why is this commented out?
Did we comment this out, or the Smithy-Go authors?

There are a few other commented out methods.

Reply:
Again, fast prototyping and I didn't wanna lose the code. Fixed this (removed) since now we are extending `SymbolWriter` instead of older `AbstractCodeWriter`.